packages/base/src/3dview/mainview.tsx

```diff
@@ -1097,10 +1115,71 @@ export class MainView extends React.Component<IProps, IStates> {
         if (boundingBox) {
           boundingBox.visible = true;
         }
+        const matchingChild = this._meshGroup?.children.find(child =>
+          child.name.startsWith(selectedMesh.name)
+        );
+        if (matchingChild && !this._clipSettings.enabled) {
+          this._transformControls.attach(matchingChild as BasicMesh);
+
+          const obj = this._model.sharedModel.getObjectByName(
+            selectedMesh.name
+          );
+
+          const positionArray = obj?.parameters?.Placement?.Position;
+          if (positionArray && positionArray.length === 3) {
+            const positionVector = new THREE.Vector3(
+              positionArray[0],
+              positionArray[1],
+              positionArray[2]
+            );
+
+            this._transformControls.position.copy(positionVector);
+          }
+          this._transformControls.setMode('translate');
+          this._transformControls.visible = true;
+          this._transformControls.enabled = true;
+        }
```

Comment:
I think we should not do any of this if there is more than one object selected. As we won't be able to translate multiple objects at once.

So this code should not be in the for loop `for (const selectionName in selection) {` but outside of it, wrapped in a `if (selection.length === 1)` check

Reply:
Done in [`0f65631`](https://github.com/jupytercad/JupyterCAD/pull/529/commits/0f656311ae9b5154dfe7da62177e66f5fe453829) thanks @martinRenou 🚀